### PR TITLE
Upgrade koa and its types to latest

### PR DESCRIPTION
**Pull request by vsumner**
## Description
This PR upgrades `koa` and `@types/koa` to the latest version.

## Type of change

- [ ]  Patch: Bug/ Documentation fix (non-breaking change which fixes an issue or adds documentation)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added a changelog entry, prefixed by the type of change noted above

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -7563,13 +7571,25 @@ mime-db@1.40.0:
...
+  integrity sha512-+5dsGEEovYbT8UY9yD7eE4XTc4UwJ1jBYlgaQQF38ENsKR3wj/8q8RFZrF9WIZpB2V1ArTVFUva8sAul1NzRzQ==
+
+mime-types@^2.1.12, mime-types@~2.1.19:
   version "2.1.24"
   resolved "https://registry.yarnpkg.com/mime-types/-/mime-types-2.1.24.tgz#b6f8d0b3e951efb77dedeca194cff6d16f676f81"
   integrity sha512-WaFHS3MCl5fapm3oLxU4eYDw77IQM2ACcxQ9RIxfaC3ooc6PFuBMGZZsYpvoXS5D5QTWPieo1jjLdAm3TBP3cQ==
   dependencies:
     mime-db "1.40.0"
 
+mime-types@^2.1.18, mime-types@~2.1.24:
```
Comment (alexandcote) [inline yarn.lock]: I think you can dedupe this one.

File: yarn.lock [hunk 2]
```diff
@@ -1486,7 +1495,12 @@
   dependencies:
     "@types/node" "*"
 
-"@types/node@*", "@types/node@>=6":
+"@types/node@*":
+  version "13.9.0"
+  resolved "https://registry.yarnpkg.com/@types/node/-/node-13.9.0.tgz#5b6ee7a77faacddd7de719017d0bc12f52f81589"
+  integrity sha512-0ARSQootUG1RljH2HncpsY2TJBfGQIKOOi7kxzUY6z54ePu/ZD+wJA8zI2Q6v8rol2qpG/rvqsReco8zNMPvhQ==
+
+"@types/node@>=6":
```
Comment (alexandcote) [inline yarn.lock]: Same here, I think you can dedupe.

Comment (vsumner) [reply yarn.lock]: Screen Shot 2020-03-10 at 2 59 10 PM

I'm concerned about bumping it from 12 to 13. Maybe an apollo memory update would be better in another pr?

**Comment by vsumner** [comment]
The fix for the promise type I pulled from https://github.com/DefinitelyTyped/DefinitelyTyped/issues/42084

File: packages/jest-dom-mocks/src/promise.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export default class Promise {
     }
 
     jest.useFakeTimers();
-    global.Promise = FakePromise;
+    global.Promise = FakePromise as any;
```
Comment (TheMallen) [inline packages/jest-dom-mocks/src/promise.ts]: Why do we need to `as any` this now but not before?

Comment (alexandcote) [reply packages/jest-dom-mocks/src/promise.ts]: I had the same question and I tried to remove it.

image

Comment (vsumner) [reply packages/jest-dom-mocks/src/promise.ts]: `@types/node` `13` bump. This was introduced in that. See https://github.com/Shopify/quilt/pull/1316#issuecomment-597284951

**Pull request merged**